File: apps/python-api/src/yourmedportal_api/lib/codes.py
import secrets

_ALPHABET = "0123456789ABCDEFGHJKLMNPQRSTUVWXYZ"


def _suffix() -> str:
    return "".join(secrets.choice(_ALPHABET) for _ in range(5))


def new_booking_code() -> str:
    return f"PN-{_suffix()}"


def new_report_code() -> str:
    return f"RF-{_suffix()}"

File: apps/api/src/lib/codes.ts
import { customAlphabet } from 'nanoid';

const suffix = customAlphabet('0123456789ABCDEFGHJKLMNPQRSTUVWXYZ', 5);

export function newBookingCode(): string {
	return `PN-${suffix()}`;
}

export function newReportCode(): string {
	return `RF-${suffix()}`;
}
